android/guava/src/com/google/common/cache/LocalCache.java

```diff
@@ -1946,24 +1945,16 @@ static class Segment<K, V> extends ReentrantLock {
       this.statsCounter = checkNotNull(statsCounter);
       initTable(newEntryArray(initialCapacity));
 
-      keyReferenceQueue = map.usesKeyReferences() ? new ReferenceQueue<K>() : null;
+      keyReferenceQueue = map.usesKeyReferences() ? new ReferenceQueue<>() : null;
 
-      valueReferenceQueue = map.usesValueReferences() ? new ReferenceQueue<V>() : null;
+      valueReferenceQueue = map.usesValueReferences() ? new ReferenceQueue<>() : null;
 
       recencyQueue =
-          map.usesAccessQueue()
-              ? new ConcurrentLinkedQueue<ReferenceEntry<K, V>>()
-              : LocalCache.<ReferenceEntry<K, V>>discardingQueue();
+          map.usesAccessQueue() ? new ConcurrentLinkedQueue<>() : LocalCache.discardingQueue();
 
-      writeQueue =
-          map.usesWriteQueue()
-              ? new WriteQueue<K, V>()
-              : LocalCache.<ReferenceEntry<K, V>>discardingQueue();
+      writeQueue = map.usesWriteQueue() ? new WriteQueue<>() : LocalCache.discardingQueue();
 
-      accessQueue =
-          map.usesAccessQueue()
-              ? new AccessQueue<K, V>()
-              : LocalCache.<ReferenceEntry<K, V>>discardingQueue();
+      accessQueue = map.usesAccessQueue() ? new AccessQueue<>() : LocalCache.discardingQueue();
```

Comment:
**Suggestion**: Simplifying the declaration of queues using diamond operators enhances readability. Ensure that this change is consistently applied across the codebase where applicable. Additionally, verify that the switch to diamond syntax does not inadvertently affect the type inference in complex scenarios, particularly with custom or generic-heavy code paths. <br> **Code Suggestion**: 
 ```
 +      // Simplified declaration with diamond operator for improved readability
 +      keyReferenceQueue = map.usesKeyReferences() ? new ReferenceQueue<>() : null;
 ```